drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1338,11 +1338,14 @@ static void uarte_nrfx_set_power_state(struct device *dev, u32_t new_state)
 			return;
 		}
 #endif
-		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPRX);
-		while (!nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_RXTO)) {
-			/* Busy wait for event to register */
+		if (nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_RXSTARTED)) {
```

Comment:
It would be good to also clear the `RXSTARTED` event here, so that the next time this code is executed, it will check if RX has been actually started again.